sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateLibraryTemplateDao.java

```diff
@@ -0,0 +1,132 @@
+package uk.ac.bbsrc.tgac.miso.persistence.impl;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.criterion.Restrictions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryTemplate;
+import uk.ac.bbsrc.tgac.miso.core.store.LibraryTemplateStore;
+import uk.ac.bbsrc.tgac.miso.core.util.DateType;
+
+@Repository
+@Transactional(rollbackFor = Exception.class)
+public class HibernateLibraryTemplateDao implements LibraryTemplateStore, HibernatePaginatedDataSource<LibraryTemplate> {
+
+  protected static final Logger log = LoggerFactory.getLogger(HibernateLibraryTemplateDao.class);
+
+  private static final String[] SEARCH_PROPERTIES = new String[] {};
+  private static final List<String> STANDARD_ALIASES = Arrays.asList();
+
+
+  @Autowired
+  private SessionFactory sessionFactory;
+
+  @Override
+  public Session currentSession() {
+    return getSessionFactory().getCurrentSession();
+  }
+
+  @Override
+  public LibraryTemplate get(long id) throws IOException {
+    return (LibraryTemplate) currentSession().get(LibraryTemplate.class, id);
+  }
+
+  public SessionFactory getSessionFactory() {
+    return sessionFactory;
+  }
+
+  @Override
+  public long create(LibraryTemplate libraryTemplate) throws IOException {
+    return (Long) currentSession().save(libraryTemplate);
+  }
+
+  public void setSessionFactory(SessionFactory sessionFactory) {
+    this.sessionFactory = sessionFactory;
+  }
+
+  @Override
+  public void update(LibraryTemplate libraryTemplate) throws IOException {
+    currentSession().update(libraryTemplate);
+  }
+
+  @Override
+  public String getFriendlyName() {
+    return "Library Template";
+  }
+
+  @Override
+  public String getProjectColumn() {
+    return "project.id";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "project.id" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)